### PR TITLE
allow account.promote_to_merchant to take a URI. Add on_behalf_of to account.debit

### PR DESCRIPTION
**Pull request by jrus**

### CODE REVIEW

File: lib/balanced/resources/account.rb [hunk 1]
```diff
@@ -38,6 +38,19 @@ def debit *args
       meta = args[3] || options.fetch(:meta) { nil }
       description = args[4] || options.fetch(:description) { nil }
       source_uri = args[5] || options.fetch(:source_uri) { nil }
+      on_behalf_of = args[6] || options.fetch(:on_behalf_of) { nil }
+
+      if on_behalf_of
+        if on_behalf_of.uri
```
Comment (mahmoudimus) [inline lib/balanced/resources/account.rb]: this needs to be `if on_behalf_of.respond_to? :uri`

Comment (jrus) [inline lib/balanced/resources/account.rb]: yep, thx

File: lib/balanced/resources/account.rb [hunk 2]
```diff
@@ -38,6 +38,19 @@ def debit *args
       meta = args[3] || options.fetch(:meta) { nil }
       description = args[4] || options.fetch(:description) { nil }
       source_uri = args[5] || options.fetch(:source_uri) { nil }
+      on_behalf_of = args[6] || options.fetch(:on_behalf_of) { nil }
+
+      if on_behalf_of
+        if on_behalf_of.respond_to? :uri
+          on_behalf_of = on_behalf_of.uri
+        end
+        if !on_behalf_of.is_a?(String)
```
Comment (mjallday) [inline lib/balanced/resources/account.rb:47]: `if !xxx.is_a_uri?` would be a cool way to do this if we're going to start adding more logic in the client libraries

**Comment by mahmoudimus** [comment]
Thanks @jrus !

**Pull request merged**